File: src/test/resources/templates/save-as-table-request.java
package mykidong.spark;

import com.cedarsoftware.util.io.JsonWriter;
import com.fasterxml.jackson.databind.ObjectMapper;
import mykidong.reflect.DynamicSparkRunner;
import mykidong.util.Log4jConfigurer;
import mykidong.util.StringUtils;
import org.apache.hadoop.conf.Configuration;
import org.apache.spark.SparkConf;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SaveMode;
import org.apache.spark.sql.SparkSession;
import org.apache.spark.sql.types.StructType;
import org.joor.Reflect;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.core.io.ClassPathResource;
import org.springframework.core.io.Resource;
import org.springframework.core.io.support.PropertiesLoaderUtils;

class SparkRunner implements mykidong.reflect.DynamicSparkRunner {
    public String run(SparkSession spark) throws Exception {
        // read parquet.
        Dataset<Row> parquetDs = spark.read().format("parquet")
                .load("/test-event-parquet");

        // create persistent parquet table with external path.
        parquetDs.write().format("parquet")
                .option("path", "hdfs://mc/test-event-parquet-table")
                .mode(SaveMode.Overwrite)
                .saveAsTable("test_parquet_dynamic_request");

        return "SUCCESS!!";
    }
}

File: src/test/resources/templates/save-as-table-request.scala
import org.apache.spark.sql.{SaveMode, SparkSession}

trait DynamicScalaSparkJobRunner {
    def run(spark: SparkSession): String
}

class SparkRunner extends DynamicScalaSparkJobRunner {
    override def run(spark: SparkSession): String = {
        // read parquet.
        val parquetDs = spark.read.format("parquet")
                .load("/test-event-parquet")

        parquetDs.show(5)

        // create persistent parquet table with external path.
        parquetDs.write.format("parquet")
                .option("path", "hdfs://mc/test-event-parquet-table")
                .mode(SaveMode.Overwrite)
                .saveAsTable("test_parquet_dynamic_request")
        "SUCCESS!!"
    }
}
scala.reflect.classTag[SparkRunner].runtimeClass

val runner = new SparkRunner()
runner.run(spark)